include/blocking_future.h

```diff
@@ -41,6 +44,22 @@ template <typename T> class BlockingFuture {
         m_data.reset();
         return item;
     }
+    std::optional<T> take(const std::chrono::seconds _max_time) {
+        std::unique_lock<std::mutex> lock(m_mutex);
+
+        std::optional<T> ret;
+
+        m_cond.wait_for(lock, _max_time, [&] { return m_data.has_value(); });
+        
+        if(m_data.has_value())
+        {
+          auto item = std::move(m_data.value());
+          m_data.reset();
+          return item;
+        }
+        return {};
```

Comment:
Change to std::nullopt. Not clear what the behavior of this constructor is to me?